### ISSUE TITLE
Add varargs of effects to `next` and `dispatch` static methods

### ISSUE THREAD

**Issue by yoelglus** [spotify/mobius#97]
This will eliminate the need to pass in a set of effects and make the code less verbose.
Will submit a PR with how this can look like.

**Comment by pettermahlen** [comment]
We considered that and discarded it in the initial design of Mobius. Instead, we picked the `Effects.effects(F... effects)` method as the convenient way of creating a set. 

The rationale for wanting a `Set` of effects is to disable any logic that depends on the execution ordering of effects. It's a rationale that we're currently debating internally, if or when it's valuable. The idea of making it impossible to write code that depends on an implicit guarantee that the framework explicitly doesn't provide (effect execution ordering) is good, but there are some cases when the only way to guarantee ordering (effect1 -> event -> effect2) is too cumbersome. So we've started thinking about how we can do that as part of starting to sketch out what a Mobius 2.0 release might look like. 

I don't think we'll want to just make a change that allows passing in non-Set-based collections of effects into Mobius 1.0, but we do want to try to understand what the impact is of enabling ordering guarantees for effect execution as a part of making other possible changes to semantics in Mobius.
